### PR TITLE
update hidapi to 0.9.0

### PR DESCRIPTION
**Pull request by Be-ing**
The original maintainer of hidapi (signal11) stopped responding
to communication for a long time, so hidapi has been forked. It is
now hosted by libusb at
https://github.com/libusb/hidapi
0.9.0 is the first stable release since the fork.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1985,16 +1985,16 @@ if(HID)
   if(HIDAPI_STATIC)
     message(STATUS "Linking internal libhidapi statically")
     add_library(mixxx-hidapi STATIC EXCLUDE_FROM_ALL)
-    target_include_directories(mixxx-hidapi SYSTEM PUBLIC lib/hidapi-0.8.0-rc1/hidapi)
+    target_include_directories(mixxx-hidapi SYSTEM PUBLIC lib/hidapi/hidapi)
```
Comment (Be-ing) [inline CMakeLists.txt:1988]: I removed the version number from the file path so we don't have to keep updating the build system when updating the library.

**Comment by Be-ing** [comment]
@Xerus2000 could you test this?

**Comment by codecat** [comment]
I can try it, but recently I haven't been able to build it myself. Is there a nightly build available somewhere?

**Comment by Be-ing** [comment]
Our build server doesn't build pull requests, only code that has been merged. We have AppVeyor running on PRs. You can try the [AppVeyor artifacts](https://ci.appveyor.com/project/mixxxdj/mixxx/builds/31994492/job/9m9ixx0wefoi2asp/artifacts) although [I'm not sure those are actually installable right now](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/testing.20on.20Windows.20with.20appveyor.20builds).

**Comment by Be-ing** [comment]
What I'm most interested in regarding Windows if there are HID devices that show in Mixxx's preferences with a bunch of weird characters in their name. IIRC we had cherry picked a patch for that into the version of hidapi we have in master now. Hopefully it is included in this new upstream release.

**Comment by Xerus2000** [comment]
I probably don't have the opportunity to test this within the next few weeks on Windows unless that is really important. I can definitely test it on Linux within the next few days.

**Comment by Be-ing** [comment]
Testing on Linux would still be helpful.

**Comment by uklotzde** [comment]
I have tested an ancient Hercules RMX DJ on Fedora with the system hidapi v0.9.0. Works.

**Comment by Be-ing** [comment]
Okay I'll merge this.

**Pull request merged**